.github/workflows/release.yaml

```diff
@@ -30,40 +30,24 @@ jobs:
 
           echo "RELEASE_TAG=${GITHUB_REF:10}" >> $GITHUB_ENV # strip off `refs/tags/` prefix
 
-      - name: Check out code
-        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # tag=v3.5.0
+      - name: checkout code
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      # - name: Calculate Go version
-      #   run: echo "go_version=$(make go-version)" >> $GITHUB_ENV
-
-      # - name: Set up Go
-      #   uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # tag=v3.5.0
-      #   with:
-      #     go-version: ${{ env.go_version }}
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          cache-dependency-path: |
+            ./go.sum
+            ./hack/tools/go.sum
```

Comment:
Unrelated. Copied this from upstream because then our diff is smaller. This can in theory help Go caching, but since the main program (the CAPA controller) isn't even built for `make release-manifests`, it probably doesn't do much.